DPPP/test/unit/tPSet.cc

```diff
@@ -0,0 +1,447 @@
+    casacore::Vector<casacore::Int> ant1(nbl);
+    casacore::Vector<casacore::Int> ant2(nbl);
+    int st1 = 0;
+    int st2 = 0;
+    for (int i=0; i<nbl; ++i) {
+      ant1[i] = st1;
+      ant2[i] = st2;
+      if (++st2 == 4) {
+        st2 = 0;
+        if (++st1 == 4) {
+          st1 = 0;
+        }
+      }
+    }
+    casacore::Vector<casacore::String> antNames(4);
+    antNames[0] = "rs01.s01";
+    antNames[1] = "rs02.s01";
+    antNames[2] = "cs01.s01";
+    antNames[3] = "cs01.s02";
+    std::vector<casacore::MPosition> antPos(4);
+    casacore::Vector<double> antDiam(4, 70.);
+    info().set (antNames, antDiam, antPos, ant1, ant2);
+    casacore::Vector<double> chanWidth(nchan, 100000);
+    casacore::Vector<double> chanFreqs(nchan);
+    indgen (chanFreqs, 1050000., 100000.);
+    info().set (chanFreqs, chanWidth);
+  }
+private:
+  virtual bool process (const DP3::DPPP::DPBuffer&) { return false; }
+  virtual void finish() {}
+  virtual void show (std::ostream&) const {}
+
+  int itsNChan, itsNCorr;
+};
+
+} // end anonymous namespace
+
+namespace DP3 {
+  namespace DPPP {
+    // This class name should match the friend in PreFlagger.
+    class TestPSet
+    {
+    public:
+      static void testNone();
+      static void testBL();
+      static void testChan();
+      static void testTime();
+      static void testMinMax();
+    };
+
+    void TestPSet::testNone()
+     {
+       TestInput* in = new TestInput(16, 8, 4);
+       DPStep::ShPtr step1(in);
+       std::cout << "testNone" << std::endl;
+       ParameterSet parset;
+       PreFlagger::PSet pset (in, parset, "");
+       pset.updateInfo (in->getInfo());
+       BOOST_CHECK (!(pset.itsFlagOnBL   || pset.itsFlagOnAmpl || pset.itsFlagOnPhase ||
+                 pset.itsFlagOnReal || pset.itsFlagOnImag ||
+                 pset.itsFlagOnAzEl || pset.itsFlagOnUV));
+     }
+
+    void TestPSet::testBL()
+    {
+      TestInput* in = new TestInput(16, 8, 4);
+      DPStep::ShPtr step1(in);
+      {
+        std::cout << "testBL 1" << std::endl;
+        ParameterSet parset;
+        parset.add ("baseline", "[rs01.*, rs02.s01]");
+        PreFlagger::PSet pset (in, parset, "");
+        pset.updateInfo (in->getInfo());
+        BOOST_CHECK (!(pset.itsFlagOnAmpl || pset.itsFlagOnPhase || pset.itsFlagOnReal ||
+                  pset.itsFlagOnImag || pset.itsFlagOnAzEl  || pset.itsFlagOnUV) &&
+                pset.itsFlagOnBL);
+        // Make sure the matrix is correct.
+        const casacore::Matrix<bool>& mat = pset.itsFlagBL;
+        BOOST_CHECK_EQUAL (mat.shape(), casacore::IPosition(2,4,4));
+        BOOST_CHECK ( mat(0,0) &&  mat(0,1) &&  mat(0,2) &&  mat(0,3));
+        BOOST_CHECK ( mat(1,0) &&  mat(1,1) &&  mat(1,2) &&  mat(1,3));
+        BOOST_CHECK ( mat(2,0) &&  mat(2,1) && !mat(2,2) && !mat(2,3));
+        BOOST_CHECK ( mat(3,0) &&  mat(3,1) && !mat(3,2) && !mat(3,3));
+      }
+      {
+        std::cout << "testBL 2" << std::endl;
```

Comment:
Should the cout statements be removed? And maybe the explicit scoping should be replaced by separate unit tests (possible with fixture) ?